### PR TITLE
Handle --knative option for current stacks as well as stacks that do not contain app-deploy.yaml

### PR DESCRIPTION
**Pull request by kewegner**
If the --knative flag is present and there is a app-deploy.yaml containing an appsodyapplication, the createKnativeService: true tag will be added to the yaml file.

If the --knative flag is not present and createKnativeService: true is in the app-deploy.yaml file the createKnativeService element will be removed.  if it is createKnativeService: false, it will remain.

If the stack does not contain an app-deploy.yaml file containing and appsodyapplication, the --knative flag will have no impact.

Unless the appsody deploy is done with a --tag, -t option the imagename will be prefixed with dev.local/<projectname>

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -78,21 +78,27 @@ generates a deployment manifest (yaml) file if one is not present, and uses it t
...
 
-		//Retrieve the project name and lowercase it
-		projectName, perr := getProjectName()
-		if perr != nil {
-			return errors.Errorf("%v", perr)
-		}
-		deployImage := projectName // if not tagged, this is the deploy image name
+		var deployImage string
+		//	:= projectName // if not tagged, this is the deploy image name
 		if tag != "" {
```
Comment (chilanti) [inline cmd/deploy.go]: Is this check still needed? We check tag earlier and set it if not set....

File: cmd/deploy.go [hunk 2]
```diff
@@ -103,26 +109,51 @@ generates a deployment manifest (yaml) file if one is not present, and uses it t
...
 		for scanner.Scan() {
+			var localFoundCreateKnativeTag bool
 			line := scanner.Text()
 			if strings.Contains(line, "applicationImage:") {
 				index := strings.Index(line, ": ")
 				start := line[0:(index + 2)]
 				line = start + deployImage
 				Info.log("Using applicationImage of: ", deployImage)
 			}
-			txtlines = append(txtlines, line)
```
Comment (chilanti) [inline cmd/deploy.go]: I would suggest the following:
```
for ... {
if strings.Contains(line, "createKnativeService") { //it's the infamous flag
 foundCreateKnativeTag = true
 if strings.Contains(line, "true") && !knative {
  //replace with false 
 } else if strings.Contains(line, "false") && knative {
  //replace with "true"
 }
} 
append()
} //end loop
//check if !found and append in that case

Comment (kewegner) [reply cmd/deploy.go]: I think that is good, one alternative might be to just check if the tag is there and then regen using the value of knative.  But the above is OK and probably simpler.

I think the bigger problem I just thought of is the append if createKnativeService is not there... Do we know that someone has not hand edited the file and added another k8s object defintiion?  I think we need to ask @neeraj-laad what the limits of editing app-deploy.yaml are.

File: cmd/deploy.go (hunk 1 above)
Comment (chilanti) [reply cmd/deploy.go]: Understood you are setting a different tag.

File: cmd/deploy.go (hunk 2 above)
Comment (kewegner) [reply cmd/deploy.go]: Changes have been made.  We will discuss the editing of app-deploy.yaml on the scrum and see if more work is needed.

**Pull request merged**